apps/android/app/src/main/java/chat/simplex/app/views/chat/SendMsgView.kt

```diff
@@ -312,6 +184,157 @@ private fun NativeKeyboard(
   }
 }
 
+@Composable
+fun RecordVoiceView(
+  recState: MutableState<RecordingState>,
+  onAudioAdded: (filePath: String, durationMs: Int, finished: Boolean) -> Unit,
+) {
+  val rec: Recorder = remember { RecorderNative(MAX_VOICE_SIZE_FOR_SENDING) }
+  DisposableEffect(Unit) { onDispose { rec.stop() } }
+  val stopRecordingAndAddAudio: () -> Unit = {
+    recState.value.filePathNullable?.let { onAudioAdded(it, rec.stop(), true) }
+    recState.value = RecordingState.NotStarted
+  }
+  if (recState.value.isStarted && (recState.value as RecordingState.Started).stopRecOnNextClick.value) {
+    val context = LocalContext.current
+    DisposableEffect((recState.value as? RecordingState.Started)?.stopRecOnNextClick?.value) {
+      val activity = context as Activity
+      if ((recState.value as? RecordingState.Started)?.stopRecOnNextClick?.value == true) {
+        // Lock orientation to current orientation because screen rotation will break the recording
+        activity.requestedOrientation = when (activity.display?.rotation) {
+          android.view.Surface.ROTATION_90 -> ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
+          android.view.Surface.ROTATION_180 -> ActivityInfo.SCREEN_ORIENTATION_REVERSE_PORTRAIT
+          android.view.Surface.ROTATION_270 -> ActivityInfo.SCREEN_ORIENTATION_REVERSE_LANDSCAPE
+          else -> ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
+        }
+      }
```

Comment:
whenever you write comment, it is usually the place where you lack a function name - create a function with a descriptive name, it'll be easier to read.

Reply:
Comment is still needed even with a dedicated function, so a dev can understand why it's really needed to lock the orientation